src/helpers/files.js

```diff
@@ -171,22 +206,24 @@ const FilesWorkspacePlugin = {
 			const vm = new View({
 				propsData: {
 					path: fileList.getCurrentDirectory(),
+					hasDescriptionFile: !!fileList.findFile(descriptionFile)
 				},
 				store,
 			}).$mount(this.el)
-
 			subscribe('files:navigation:changed', () => {
 				// Expose if the default file list is active to the component
 				// to only render the workspace if the file list is actually visible
 				vm.active = OCA.Files.App.getCurrentFileList() === fileList
 			})
-
 			fileList.$el.on('urlChanged', data => {
 				vm.path = data.dir.toString()
 			})
 			fileList.$el.on('changeDirectory', data => {
 				vm.path = data.dir.toString()
 			})
+			fileList.$table.on('DOMSubtreeModified', () => {
+				vm.hasDescriptionFile = !!fileList.findFile(descriptionFile)
```

Comment:
I'd prefer to have an event that is triggered through https://github.com/nextcloud/nextcloud-event-bus on the newFileAction after creating the file. The RIchWorkspace.vue component could listen to that and reuse the existing way to check for a readme file through the backend https://github.com/nextcloud/text/blob/master/src/views/RichWorkspace.vue#LL145

Reply:
I definitely agree with that. There is a problem though. When I tried, I noticed that the UI takes a long time before updating the rich workspace block. I guess because of the event-bus.

Reply:
It should take the same amount of time as it is currently. There is an extra request but that should not change from how it is right now. Event bus in general should be fast ;)

Reply:
The problem is `DOMSubtreeModified` is triggered almost all the time so it's always triggering a new event (either `Text::showRichWorkspace` or `Text::hideRichWorkspace`)

Reply:
Yes, I think that is triggered way too often, therefore I'd go with the above mentioned event bus approach.

Reply:
The event bus approach is already implemented. The actual problem is the right event to use. We're using `DOMSubtreeModified` right now.

Reply:
@juliushaertl Is there any event we can listen to when the README.md file is deleted?

Reply:
@juliushaertl Could you please check again?

Reply:
Nothing yet. We could skip this for now as I know there is some ongoing work in the files2vue migration.

Otherwise a server PR with triggering an event around here would be the way to go I think:

https://github.com/nextcloud/server/blob/5b9a8f0407a9e3fe7e00f0fc9284ea986905f1b5/apps/files/js/filelist.js#L2494

Reply:
Something like: `this.$el.trigger(jQuery.Event('removeFile', path));`